dbmongo/js/reduce.algo2/lookAhead.ts

```diff
@@ -16,11 +16,11 @@ export function lookAhead(
 
   /* eslint-disable */
   var sorting_fun = function (a: any, b: any): any {
-    return a >= b
+    return a >= b ? 1 : -1 // TODO: normally, a sorting comparator should return a number, possibly including zero. => the TS version of the test has failed until we added `? 1 : -1` here
```

Comment:
Hypothèse:
- JSC et Node.js n'ont pas la même manière de ranger les éléments (`sort()`)
- Pour réduire le risque de résultats différents, il faut au moins que cette fonction retourne `0` dans le cas où les deux éléments comparés sont égaux, cf https://stackoverflow.com/a/55995920/592254

Ce n'est pas le seul endroit du code où j'ai observé des fonctions de comparaison qui ne renvoient que deux valeurs possibles (c.a.d. le résultat booléen d'une inégalité entre les deux éléments comparés), alors que cette fonction est sensée retourner un nombre avec au moins 3 valeurs possibles: -1, 0 et 1.

=> Nous devrions prendre le temps de définir plus précisément ces tris, et ajuster les tests impactés, si nécéssaire.

Reply:
Je comprends que typescript râle si on lui passe un booléen plutôt qu'un integer. 
Ce qui est plus étonnant, c'est que lorsque casté, le booléen donne 0 et 1 là où il devrait donner -1 et 1... Bug ?

Reply:
Effectivement, TS râle si on retourne un boolean. Mais même si on ignore ce warning, le comportement de ce code est différent sur Node.js et JSC => les tests échouaient jusqu'à ce que j'ajoute `? 1 : -1` pour transformer le resultat booléen en nombre !

Il faudrait repenser ce comparateur de manière à ce qu'il puisse également retourner 0 dans le cas ou `a` et `b` sont égaux. Mon hypothèse est qu'en faisant ça, on gommerait la différence de comportement entre JSC et Node.js, les tests passeraient dans les deux cas, et le tri serait probablement plus performant.